### PR TITLE
Added Exception "Hotel with bookings greater than 0 can not be deleted"

### PR DESCRIPTION
**Pull request by aryan-dua**
Code Reviewer Checklist:
- [ ] Tech Documentation : On Knowledge/ Google Doc
- [ ] Proper formatting, indentation, No spelling mistakes
- [ ] Meaningful Variable, Function, Class Names etc.
- [ ] Proper usage of access modifiers - private, public etc
- [ ] No Duplicate Code
- [ ] PR should be less than 10 files
- [ ] Classes should be less than 200 lines
- [ ] Functions should be less than 20 lines
- [ ] Code Documentation
- [ ] Test Cases
- [ ] Local Functions (Function Group) optimized for space, time & resources used
- [ ] No unnecessary and unused variables
- [ ] No unnecessary database calls in code (Cache usage if possible)
- [ ] Proper Error Handling in place
- [ ] Proper Logging (such a way that it is easier to find it on Kibana)

### CODE REVIEW

File: src/main/java/com/example/hbs/service/HotelService.java [hunk 1]
```diff
@@ -98,6 +103,11 @@ public HotelResponseDto deleteHotel(Long id, Long userId) {
         Hotel hotel = hotelRepository.findById(id).orElse(null);
         Optional.ofNullable(hotel).orElseThrow(() -> new HbsException("Hotel not found"));
         checkIfUserIsOwner(hotel, userId);
+        Pageable page = PageRequest.of(0, 10);
+        List<Booking> booking = bookingRepository.findAllBookingofHotel(id, page).getContent();
+        if(!booking.isEmpty()){
+            throw new HbsException("Hotel with bookings greater than 0 can not be deleted");
+        }
```
Comment (harisoyo) [inline src/main/java/com/example/hbs/service/HotelService.java]: hotel.getBookings should work

File: src/main/java/com/example/hbs/service/HotelService.java [hunk 2]
```diff
@@ -109,7 +119,9 @@ public HotelResponseDto updateHotel(Long id, HotelResponseDto hotelResponseDto,
         Optional.ofNullable(hotel).orElseThrow(() -> new HbsException("Hotel not found"));
         checkIfUserIsOwner(hotel, userId);
         if (hotelResponseDto.getAvailableRooms() != null) {
+            Integer roomsAdded = hotelResponseDto.getAvailableRooms() - hotel.getAvailableRooms();
             hotel.setAvailableRooms(hotelResponseDto.getAvailableRooms());
+            hotel.setNoOfRooms(hotel.getNoOfRooms() + roomsAdded);
```
Comment (harisoyo) [inline src/main/java/com/example/hbs/service/HotelService.java]: logic?

**Review by harisoyo — CHANGES_REQUESTED** [review]

File: src/main/java/com/example/hbs/service/HotelService.java (hunk 2 above)
Comment (aryan-dua) [reply src/main/java/com/example/hbs/service/HotelService.java]: If we are adding or subtracting Available rooms without adding or deleting bookings So no of rooms should also be increased/decreased accordingly, right?

File: src/main/java/com/example/hbs/service/HotelService.java (hunk 1 above)
Comment (aryan-dua) [reply src/main/java/com/example/hbs/service/HotelService.java]: Okay

Comment (aryan-dua) [reply src/main/java/com/example/hbs/service/HotelService.java]: I think this is same, just my function name is different

**Review by harisoyo — APPROVED** [review]

**Pull request merged**